ucrop/src/main/java/com/yalantis/ucrop/util/BitmapLoadUtils.java

```diff
@@ -50,14 +50,10 @@ public static Bitmap decode(@NonNull Context context, @Nullable Uri uri,
             close(parcelFileDescriptor);
         }
 
-        ExifInterface exif = getExif(uri);
-        if (exif != null) {
-            int exifOrientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL);
-            // TODO Should not rotate bitmap but initially apply needed angle to the matrix
-            return rotateBitmap(decodeSampledBitmap, exifToDegrees(exifOrientation));
-        } else {
-            return decodeSampledBitmap;
-        }
+
+        int exifOrientation = getExifOrientation(context, uri);
+        // TODO Should not rotate bitmap but initially apply needed angle to the matrix
```

Comment:
You can apply a scale of -1 to the matrix to get the mirrored orientations. See the logic for static Bitmap transformResult(Request data, Bitmap result, int exifOrientation) at https://github.com/square/picasso/blob/master/picasso/src/main/java/com/squareup/picasso/BitmapHunter.java.

There's also some logic for recalculating the height and width depending on the orientation.